### DISCUSSION TITLE
Custom order of DataObjects in Pimcore

### DISCUSSION THREAD

**Discussion by lostdesign** [pimcore/pimcore#9435] [Q&A / Support]
Hi there, I've seen that you can order your objects manually by "index". Can anyone elaborate what that means? Is there any way to set a custom value in order to get a particular order in the tree?

![example](https://user-images.githubusercontent.com/13643455/71905382-57f11a80-3168-11ea-9eed-573831587f67.png)

**Comment by valentin-** [comment]
Hi, you should be able to sort the items (childs) using drag'n'drop after the 2nd option has been applied to the parent item

**Reply by lostdesign** [reply]
@valentin- I should have been bit more detailed with my question.

We create data objects programmatically and wanted to set them in a specific order upon creation. Turns out that you can simply use `->setIndex()` in order to set the position, as that is also what the drag'n'drop does.

We set the default to `Index (Ordered By Manually)` for all our DataObjects overriding `AbstractObject::getChildrenSortBy()` using traits. Perhaps there is a better than that?

Can you configure the default "Sort By Children" ?

**Comment by stefanos87** [comment]
This feature is convenient because with drag and drop I can change the sorting manually, the problem is that when I have many objects, these are paginated, how can I move for example an object from the last page to the first?

![image](https://user-images.githubusercontent.com/21968830/159271117-67492fc4-6524-4275-a579-85b92a13e779.png)

Thank you

**Reply by Niklasbrandstrom** [reply]
This is something i have thought about to -  right now it's look like you can't do it at all. 
![signed-url-redirect](https://github.com/pimcore/pimcore/assets/97362768/af209c16-f9af-434b-97ff-184d7eda7a54)

**Reply by twin-elements** [reply]
@Niklasbrandstrom @stefanos87 Have you found a solution for this problem?

**Reply by Niklasbrandstrom** [reply]
@twin-elements Sorry to say, But no :(

**Reply by twin-elements** [reply]
@Niklasbrandstrom And this is a serious problem...
Thank you for your reply.

**Comment by twin-elements** [comment]
Has anyone researched this topic? Are there any ready-made plugins for manual changing indexes with more elements in the tree?

**Reply by markus-moser** [reply]
There is no easy option to move between pages but when you do not have too many pages there would be the option to change the `tree_paging_limit` to a higher value then the default one. The setting is something like this:

```
pimcore:
    object:
          tree_paging_limit: 100
```

With this it would be possible to avoid paging. But for sure it depends on the number of items per folder if this could work.

**Reply by twin-elements** [reply]
@markus-moser You saved me! Thanks!
